src/app/components/daily-transaction-piechart/daily-transaction-piechart.component.ts

```diff
@@ -1,90 +1,80 @@
+    responsive: true,
+    maintainAspectRatio: false,
+    plugins: {
+      legend: {
+        display: true,
+        position: 'top',
+      },
+    },
+  };
+
+  public selectedDate: string = new Date().toISOString().split('T')[0];
 
   ngOnInit() {
-    this.updateChart()
-  }
-
-
-  clearDate() {
-    this.selectedDate = '';
     this.updateChart();
   }
-  updateChart() {
-
-    this.pieChartLabels = [];
-    this.pieChartData = [];
-
-    if (this.selectedDate) {
 
-      const selectedDateObject = new Date(this.selectedDate);
-      const selectedDateTransactions = this.transactions.filter((transaction: any) => {
-        const transactionDate = new Date(transaction.transaction_date);
+  updateChart() {
+    this.pieChartData.datasets[0].data = [];
 
+    const selectedDateObject = new Date(this.selectedDate);
+    const selectedDateTransactions = this.transactions.filter(
+      (transaction: TransactionComponent) => {
+        const transactionDate = new Date(transaction.transactionDate);
         selectedDateObject.setHours(0, 0, 0, 0);
         transactionDate.setHours(0, 0, 0, 0);
-
         return selectedDateObject.getTime() === transactionDate.getTime();
-      });
-      const typeData: { [type: string]: number } = {};
-
-      selectedDateTransactions.forEach((transaction: any) => {
-        const type = transaction.transaction_type;
-        if (typeData[type]) {
-          typeData[type] += transaction.amount;
-        } else {
-          typeData[type] = transaction.amount;
-        }
-      });
-
-      const transactionTypes = Object.keys(typeData);
-      const totalAmounts = transactionTypes.map((type) => typeData[type]);
-
-      this.pieChartLabels = transactionTypes as Label[];
-      this.pieChartData = totalAmounts;
-    } else {
-
-      const typeData: { [type: string]: number } = {};
-
-      this.transactions.forEach((transaction: any) => {
-        const type = transaction.transaction_type;
-        if (typeData[type]) {
-          typeData[type] += transaction.amount;
-        } else {
-          typeData[type] = transaction.amount;
-        }
-      });
-
-      // Extract transaction types and corresponding total amounts
-      const transactionTypes = Object.keys(typeData);
-      const totalAmounts = transactionTypes.map((type) => typeData[type]);
-
-      this.pieChartLabels = transactionTypes as Label[];
-      this.pieChartData = totalAmounts;
-
-
-
-    }
-
-    this.chart.chart.update()
+      }
+    );
+
+    const typeData: { [type: string]: number } = {
+      Deposit: 0,
+      Withdrawal: 0,
+      Transfer: 0,
+      Credit: 0,
+    };
+
+    selectedDateTransactions.forEach((transaction: TransactionComponent) => {
+      let type = transaction.transactionType.slice(5);
+      type = type.charAt(0).toUpperCase() + type.slice(1).toLowerCase();
+      if (
+        type === 'Transfer' &&
+        transaction.targetAccountNumber ===
+          TransactionComponent.getAccountNumberFromToken()
```

Comment:
for reading account number, decoding token again and again, not good practice

Reply:
Thanks for letting me know. I will fix it in another PR.